1000$

```diff
@@ -1,62 +1,91 @@
+            int absHuman = std::abs(sum(human) - randNumber);
+
             std::cout << "Game begin" << std::endl
-                      << "Pc chose: " << computerChoice << std::endl
-                      << "Your choice: " << humanChoice << std::endl
-                      << "Number: " << randNumber << std::endl;
+                      << "Pc chose: " ; pc.showNumbers();
+            std::cout <<  "Your choice: "; human.showNumbers();
+            std::cout << "Number: " << randNumber << std::endl;
+
             if (absComputer == absHuman) {
-                defineWinnerByRange(humanChoice, computerChoice);
+                defineWinnerByRange();
             } else {
-                defineWinnerByAbs(absHuman, absComputer);
+                defineWinnerByAbs(absComputer, absHuman);
             }
-            balance.showBalance();
+            human.getScore();
+            pc.getScore();
+        }
     }
+
     int getPlayerInput(){
-        int number;
-        std::cout << "Please enter the number between 1 - 10 range:";
-        while(std::cin >> number && !(number >= 1 && number <= 10)){
-            std::cout << "Invalid input. Repeat.";
-        }
-        return number;
+            std::cout << "Please enter number between 1 - 20 range:";
+            int number;
+            while (std::cin >> number && !(number >= 1 && number <= 20)) {
+                std::cout << "Invalid input. Repeat.";
+            }
+            return number;
+    }
+
+    int calcComputerTurn(){
+        int pc_value;
+        pc_value = randomEngine.getRandomNumber();
+        if(pc_value )
+        return pc_value;
     }
-    int calcComputerTurn(int humanNumb){
-        computerChoice = randomService.getRandomNumber();
-        while(computerChoice == humanNumb){
-            computerChoice = randomService.getRandomNumber();
+
+    int sum(const Player &player){
+        int sum = 0;
+        const std::vector<int>& nums = player.getNumbers();
+        for (auto it = nums.begin(); it != nums.end(); ++it) {
+            sum += *it;
         }
-        return computerChoice;
+        return sum;
     }
-    void defineWinnerByRange(int compChoice, int humChoice){
-        if(humChoice < compChoice){
+
+    void defineWinnerByRange(){
+        if(human.findMinValue() < pc.findMinValue()){
             std::cout << "U R win because your number were less" << std::endl;
-            balance.humWinBalance();
+            human.winBalance();
+            pc.loseBalance();
         }else {
             std::cout << "PC win because his number were less" << std::endl;
-            balance.pcWinBalance();
+            pc.winBalance();
+            human.loseBalance();
         }
     }
+
     void defineWinnerByAbs(int humAbs, int compAbs){
-        if(humAbs < compAbs){
+        if(humAbs > compAbs){
             std::cout << "U R win because your number were closer" << std::endl;
-            balance.humWinBalance();
+            human.winBalance();
```

Comment:
Image situation.

For example,

You have balance in data base.

In this code, you have two transaction.

One for minus balance,
Second for plus balance.

If something wrong, one get minus and nobady get plus.

For avoid this issue, we have to make it in one transaction.

transaction start
... new value for one user
... new value for another user
transaction end.

But in this case you have two different method, make transaction wrapp on this inreal.

Let's make new service 'Payment'
let's method 'transfer' (out player, in player) {
and make this logic here
} 

In future we can wrap this method in transaction. like
@Transaction
'transfer' (out player, in player) {
and make this logic here
}